rfcs/0003-coverage-identifier-search.md

```diff
@@ -0,0 +1,127 @@
+# RFC Proposal
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0003-coverage-identifier-search`
+* Start Date: 2019-11-15
+* RFC PR: []()
+* JIRA Ticket(s):
+    * [BLUEBUTTON-1517](https://jira.cms.gov/browse/BLUEBUTTON-1517)
+
+Search by Coverage.identity, initially only supporting the part D coverage identities.
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+    * [Proposed Solution: Detailed Design](#proposed-solution-detailed-design)
+    * [Proposed Solution: Unresolved Questions](#proposed-solution-unresolved-questions)
+    * [Proposed Solution: Drawbacks](#proposed-solution-drawbacks)
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+## Motivation
+[Motivation]: #motivation
+
+To increase the usability of the API
+provided by the BFD
+by enabaling searching for beneficiaries
+by certain attributes.
+In this case
+their Part D entrollment status.
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+### Patient centric interface
+
+`v1.1/fhir/Patient/?ptdcntrct<month_code>=<contract id>`
+
+returns a [bundle](https://www.hl7.org/fhir/bundle.html) of [patient resources](https://www.hl7.org/fhir/patient.html)
+that have that `contract_id` for that `month_code`.
+
+
```

Comment:
I like the `_tag` parameter.

Is using [chained resource references](https://hapifhir.io/doc_rest_operations.html#_toc_chained_resource_references) another option? I think the resulting call would look something like `Patient/?Coverage.identifier=https://bluebutton.cms.gov/resources/variables/ptdcntrct01|<contract_id>`

Reply:
Interesting. I think you are on to something which I didn't think about. Cool. Yes. I think the tag search could be chained. 

As I read the spec, the patient resource search would be:
  `<base_url>/Patient/?_tag=partdcontract-01-1234`

So the chained coverage resource search would be:
  `<base_url>/Coverage/?Beneficiary._tag=partdcontract-01-1234`

Assuming that the prefix of the tag `partdcontract`. You can choose the prefix you want. (PS Why don't you like vowels?)

Reply:
Oh. Reading your suggestion again. I don't think you can use Coverage to Patient chaining because Coverage is not a field of Patient. However, both EOB and Coverage do have a beneficiary or patient field so the chain can go that way (see my suggestion above).

Reply:
One more thought. There may be scale / timeout issues with the suggestion that I had. If you supported EOB chaining:
`<base_url>/ExplainationOfBenefit/?Beneficiary._tag=partdcontract-01-1234`

The result would be huge bundle because the single patient EOB bundle is huge. This would include many patients. Definitely would run into time out and SLA issues.

Reply:
One more option/twist there's the idea of ["Reverse Chaining"](https://www.hl7.org/fhir/search.html#has). Which would correct the issue of a Patient not having a coverage property but allow this first interface to remain Patient centric.

the call would look like: ` <base_url>/Patient/?_has:Coverage.identifier=https://bluebutton.cms.gov/resources/variables/ptdcntrct03|<contract_id>`